python_modules/dagster/dagster/core/definitions/output.py

```diff
@@ -64,35 +62,45 @@ class OutputDefinition:
             partitions from the OutputContext) which should be associated with this OutputDefinition.
     """
 
+    dagster_type: DagsterType
```

Comment:
This would make these attributes mutable, which is undesirable.

Reply:
reverted